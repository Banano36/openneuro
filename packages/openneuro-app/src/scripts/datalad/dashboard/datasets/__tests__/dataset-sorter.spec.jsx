import React from 'react'
import { shallow } from 'enzyme'
import { SortField } from '../dataset-sorter.jsx'

describe('dashboard/datasets/DatasetSorter', () => {
  describe('SortField', () => {
    it('renders successfully', () => {
      expect(
        shallow(<SortField field="created" queryVariables={{ orderBy: {} }} />),
      ).toMatchSnapshot()
    })
    it('calls refetch on click', () => {
      const refetch = jest.fn()
      const wrapper = shallow(
        <SortField
          field="created"
          queryVariables={{ orderBy: {} }}
          refetch={refetch}
        />,
      )
      wrapper.simulate('click')
      expect(refetch).toHaveBeenCalled()
    })
    it('switches ascending to descending field when clicked', () => {
      const refetch = jest.fn()
      const wrapper = shallow(
        <SortField
          field="created"
          queryVariables={{ orderBy: { created: 'ascending' } }}
          refetch={refetch}
        />,
      )
      wrapper.simulate('click')
      expect(refetch).toHaveBeenCalledWith({
        orderBy: { created: 'descending' },
      })
    })
    it('switches to a new sort field when clicked', () => {
      const refetch = jest.fn()
      const wrapper = shallow(
        <SortField
          field="stars"
          queryVariables={{ orderBy: { created: 'ascending' } }}
          refetch={refetch}
        />,
      )
      wrapper.simulate('click')
      expect(refetch).toHaveBeenCalledWith({ orderBy: { stars: 'descending' } })
    })
    it('preserves unrelated fields when clicked', () => {
      const refetch = jest.fn()
      const wrapper = shallow(
        <SortField
          field="created"
          queryVariables={{
            orderBy: { created: 'ascending' },
            filterBy: { public: true },
          }}
          refetch={refetch}
        />,
      )
      wrapper.simulate('click')
      expect(refetch).toHaveBeenCalledWith({
        orderBy: { created: 'descending' },
        filterBy: { public: true },
      })
    })
  })
})
