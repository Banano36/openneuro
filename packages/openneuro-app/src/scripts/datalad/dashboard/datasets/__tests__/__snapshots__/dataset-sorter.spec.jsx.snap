// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard/datasets/DatasetSorter SortField renders successfully 1`] = `
<a
  className="btn-sort name"
  key="created"
  onClick={[Function]}
>
  <Styled(span)>
    created
  </Styled(span)>
   
</a>
`;
